### PR TITLE
feat: hosting static jsonld files

### PR DESCRIPTION
**Pull request by dvasunin**
## Description

The PR contains one feature and a small refactoring:
- files placed to the resources/verifiablecredentials directory are served by the service's web-server as static 
resources. If a file has .jsonld extension then correct `Content-Type` is set for it even if extension is missed in URL
(e.g. for URL https://{SERVICE_HOST}/context/sd-document-v2210 the file resources/verifiablecredentials/sd-document-v2210.jsonld
is returned while `Content-Type` is set to `application/ld+json`)
- unused configuration entries in application.yml are not mandatory (e.g. `app.usersDetails.custodianWallet`
for catena-x-ctx profile)
- if `test` profile is active then authentication is turned off for the service

## Why:

SD-Factory produces JSON-LD files where context definition needs to be published somewhere. Before external resources were used to host these files. With this PR context definition files can be hosted right by the SD-Factor's web-service. Before it was not possible as the builtin web-service returned wrong context-type for json-ld.

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Review by sachinargade123 — APPROVED** [review]
LGTM

**Review by ChetanT-System — APPROVED** [review]
Looks good to me !

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -45,18 +45,10 @@ app:
     version: ^project.version^
   verifiableCredentials:
     durationDays: 90
-    schema2210Url: https://f1c82785-5598-41c7-a083-01a8e1a80e19.mock.pstmn.io/ctxsd
+    schema2210Url: https://sdfactory.int.demo.catena-x.net/context/sd-document-v2210
     gaia-x-participant-schema: https://registry.lab.gaia-x.eu/development/api/trusted-schemas-registry/v2/schemas
     gaia-x-service-schema: https://gaia-x.gitlab.io/gaia-x-community/gaia-x-self-descriptions/service/ontology.json
```
Comment (tomaszbarwicki) [inline src/main/resources/application.yml:50]: hey @dvasunin, we should avoid consortia related references (urls, environments etc.). Would also strongly suggest to leverage helm values.yaml for apps configuration of such environment related variables adhereing to [TRG 5.06](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-06)

**Review by tomaszbarwicki — CHANGES_REQUESTED** [review]

File: src/main/resources/application.yml (hunk 1 above)
Comment (adityagajbhiye9) [reply src/main/resources/application.yml:50]: @tomaszbarwicki  we are looking into it.

Comment (sachinargade123) [reply src/main/resources/application.yml]: @dvasunin Remove this hardcoded URL. I will suggest left as empty value and comment over for user information.

File: src/main/java/org/eclipse/tractusx/selfdescriptionfactory/config/SecurityConfig.java [hunk 2]
```diff
@@ -85,31 +88,41 @@
...
-        http.cors().configurationSource(corsConfigurationSource());
+        // Configure stateless session management
+        http.sessionManagement(session -> session.sessionCreationPolicy(
+                SessionCreationPolicy.STATELESS
+        ));
 
-        // State-less session (state in access-token only)
-        http.sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS);
+        // Disable CSRF due to stateless session management
+        http.csrf(AbstractHttpConfigurer::disable);
```
[the remark this answers is not in the document]
Comment (dvasunin) [reply src/main/java/org/eclipse/tractusx/selfdescriptionfactory/config/SecurityConfig.java:110]: Disable CSRF due to stateless session management

**Comment by dvasunin** [comment]
@tomaszbarwicki the CodeCL scanning reports an issue which is false positive from my perspective. We disabled CSRF for purpose and it does not have any impact on application security. If you can ignore the warning and merge the PR, please do it. If not then I better to ask on DevSecOps how to proceed if we have such situations.

File: src/main/resources/application.yml (hunk 1 above)
Comment (dvasunin) [reply src/main/resources/application.yml]: done

**Review by adityagajbhiye9 — APPROVED** [review]
LGTM

**Comment by tomaszbarwicki** [comment]
hey @RoKrish14, since it's security related check, can you please review/assist from your end?

**Comment by tomaszbarwicki** [comment]
> @tomaszbarwicki the CodeCL scanning reports an issue which is false positive from my perspective. We disabled CSRF for purpose and it does not have any impact on application security. If you can ignore the warning and merge the PR, please do it. If not then I better to ask on DevSecOps how to proceed if we have such situations.

hey @RoKrish14, since it's security related check failing, can you please review/assist from your end?

**Review by RoKrish14 — APPROVED** [review]
CodeQL issue: Disabled Spring CSRF protection - marked as FP

**Review by tomaszbarwicki — APPROVED** [review]

**Pull request merged**